### ISSUE TITLE
Spectra dsn import/export

### ISSUE THREAD

**Issue by NavinF** [tscircuit/tscircuit#322]
Are there any plans to support dsn -> tscircuit soup -> dsn? AFAIK that's the de facto workflow for EDA tools and necessary for a future tscircuit autorouter to be useful

**Comment by seveibar** [comment]
@NavinF yes we have started work here and we'll probably complete it very soon.

This repo has the converter: https://github.com/tscircuit/specctra-dsn-json

CC @andrii-balitskyi

**Comment by seveibar** [comment]
@andrii-balitskyi this might be a good thread to discuss the progress on that project, are there any large remaining items? I know we got pretty far on parsing the dsn files within the freerouting project.

@imrishabh18 maybe you could also show an example of `circuit-to-png` w/ snapshot testing in that repo, that would probably help us track progress/regressions

**Comment by andrii-balitskyi** [comment]
@seveibar There's a draft PR that adds support for converting DSN JSON into tscircuit soup. This PR was put on hold as our focus shifted to another project. Additionally, the parsing of DSN files hasn't been thoroughly tested, so there may be some inaccuracies in the produced DSN JSON files.

**Comment by imrishabh18** [comment]
`specctra-dsn-json` lib doesn't give the soup it only converts it to json, right?

**Comment by seveibar** [comment]
@imrishabh18 yea it needs that draft PR for DSN JSON to tscircuit soup before we can do snapshots, but the code to convert to soup will still stay inside the `specctra-dsn-json` repo.
